src/simplifier.cpp

```diff
@@ -1058,6 +1058,15 @@ static size_t countTriangles(Triangle* tritable, size_t tritable_size, const Ver
 		VertexID b2 = vertex_ids_pos_only[indices[i + 1]];
 		VertexID c2 = vertex_ids_pos_only[indices[i + 2]];
 
+		// If one of the vertices in this triangle is locked, add 1 and continue to the next.
```

Comment:
I think we came to the conclusion that this might not quite be correct, but the worst case is that it doesn't necessarily decimate as far as it could in some cases. It should still produce correct results though, there might just be some edge cases where the decimation could be a little better because it hits some threshold. It's very hard to determine at this point what to do though at this stage at the algorithm (and we're not sure if it's possible tbh the way it's implemented here since it doesn't determine which vertices to merge/preserve until later.)

Specifically if all three vertices are in the same lod cell then it would normally merge them, but if three are locked it still needs to produce a triangle since it can't merge them. If two are locked, it may still produce a triangle assuming the third one isn't merged into one of the locked two, and so on. There might be more cases where the locked vertices are in a different grid cell etc.

Conclusion is that we don't need to address this for now since it seems like it should produce correct results anyway and we haven't seen any defects in testing, but Codie is going to document it for future reference.